frontend/tests/unit/store.index.spec.js

```diff
@@ -0,0 +1,156 @@
+//
+// Copyright (c) 2019 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Vuex from 'vuex'
+import { expect } from 'chai'
+import { state, actions, getters, mutations, modules } from '@/store'
+import find from 'lodash/find'
+
+describe('Store', function () {
+  it('should transform machine images from cloud profile', function () {
+    const cpMachineImages = [
+      {
+        'name': 'coreos',
+        'versions': [
+          {
+            'version': '2135.6.0'
+          }
+        ]
+      },
+      {
+        'name': 'suse-jeos',
+        'versions': [
+          {
+            'version': '15.1.20190927'
+          },
+          {
+            'version': '15.1.20191027',
+            'expirationDate': '2119-04-05T01:02:03Z' // not expired
+          },
+          {
+            'version': '15.1.20191127',
+            'expirationDate': '2019-04-05T01:02:03Z' // expired
+          }
+        ]
+      },
+      {
+        'name': 'foo',
+        'versions': [
+          {
+            'version': '1.02.3' // invalid version (not semver compatible)
+          },
+          {
+            'version': '1.2.3'
+          }
+        ]
+      }
+    ]
+
+    state.cloudProfiles.all = [
+      {
+        data: {
+          machineImages: cpMachineImages
+        },
+        metadata: {
+          name: 'foo'
+        }
+      }
+    ]
+
+    const store = new Vuex.Store({
+      state,
+      actions,
+      getters,
+      mutations,
+      modules
+    })
+
+    const dashboardMachineImages = store.getters.machineImagesByCloudProfileName('foo')
```

Comment:
use getter directly

Reply:
not sure how to achieve this, need to discuss this on monday

Reply:
done